client/build.go

```diff
@@ -0,0 +1,97 @@
+package client
+
+import (
+	"context"
+
+	"github.com/moby/buildkit/client/buildid"
+	gateway "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/moby/buildkit/frontend/gateway/grpcclient"
+	gatewayapi "github.com/moby/buildkit/frontend/gateway/pb"
+	"github.com/moby/buildkit/session"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+type build struct {
+	id string
+	s  *session.Session
+	c  gateway.Client
+}
+
+func (c *Client) Build(ctx context.Context, opt SolveOpt, build gateway.BuildFunc, statusChan chan *SolveStatus) (*SolveResponse, error) {
+	defer func() {
+		if statusChan != nil {
+			close(statusChan)
+		}
+	}()
+
+	if opt.Frontend != "" {
+		return nil, errors.New("invalid build opts cannot use frontend")
+	}
+
+	feOpts := opt.FrontendAttrs
+	opt.FrontendAttrs = nil
```

Comment:
I decided it was better to reuse `SolveOpt` with the switch to using `Solve` instead of `Create`/`Delete`, it results in this little wrinkle but I think that's tollerable compared with copying every field of a `BuildOpt` into a `SolveOpt`.